### ISSUE TITLE
Move some stuff to NuGet

### ISSUE THREAD

**Issue by harry-cpp** [MonoGame/MonoGame#5262]
So... here we go:
- It would be much better to not add local assemblies during the install and instead make the default templates reliant on NuGet
  - a problem that will occur for anyone transitioning from 3.5 to 3.6 is that they need to recreate the project to get the SDL native libs
  - we could package all the native libs inside the NuGet and instead of adding them to the project we could create a .targets file that automatically copy them to the output folder, therefore the end users won't even notice any difference between future versions
- we could also create a Universal MGCB.exe assembly in each package or a separate package
  - for the most part we just need to replace few "#ifs" with runtime detection
  - targets file should be in this NuGet package
  - this would make it so the end user doesn't even need to install anything besides a NuGet package or 2 to get started
- we could also package Pipeline Tool, but I would rather leave it as a system install
  - we should however make it check the current .mgcb files project for a NuGets MGCB.exe to use instead of the system one

In short:
- Make MGCB universal
- Package MGCB inside NuGet packages, or create a separate one for it?
- Package native libs inside each platforms NuGet package
- Make templates reliant on NuGet stuff
- Remove local assemblies and .targets file from the installers

**Comment by Jjagg** [comment]
I'm all for this! Especially distributing native libs and .targets through NuGet.

**Comment by cra0zy** [comment]
@tomspilman @KonajuGames @dellis1972 Thoughts on this?

**Comment by DDReaper** [comment]
I'm certainly up for this.  I did suggest it when we got the NuGet system introduced but it was felt we needed to ensure it was stable first.

Now that even .NetCore is NuGet only, this would be a good move.  Although there is a separate thread for NuGetting the MGCB tool, which has it's own challenges.

**Comment by cra0zy** [comment]
> Although there is a separate thread for NuGetting the MGCB tool, which has it's own challenges.

I got MGCB from NuGet working pretty easily (tested on Windows and Linux, in theory it should work on Mac)... just needed to connect all the files in definition from the start and replace few preprocessor platform detections with runtime platform detections...

**Comment by DDReaper** [comment]
That's interesting @cra0zy and saves me a ton of time.  Do you have a PR done for it yet and tested on a clean machine?

**Comment by cra0zy** [comment]
> Do you have a PR done for it yet and tested on a clean machine?

I'll notify you when I make a PR, I will not have time to work on it for another 2 weeks, also it will be in 2 peaces, one for making MGCB universal and another one for the NuGet package.

**Comment by DDReaper** [comment]
Fair enough @cra0zy 

Although I'm now looking at another issue with the NuGet's for Universal (possibly other platforms as well), as the dev NuGets don't work 🈂️

**Comment by cra0zy** [comment]
Just to clarify, when I say make MGCB universal, I mean make it run on Windows/Linux/Mac, not make it UWP.

**Comment by harry-cpp** [comment]
This is done, so closing.
